src/jit/optimizer.cpp

```diff
@@ -3304,7 +3304,7 @@ bool Compiler::optReachWithoutCall(BasicBlock* topBB, BasicBlock* botBB)
 
     // We can always check topBB and botBB for any gc safe points and early out
 
-    if ((topBB->bbFlags | botBB->bbFlags) & BBF_GC_SAFE_POINT)
+    if ((topBB->bbFlags & BBF_GC_SAFE_POINT) || (!endsWithTailCall && (botBB->bbFlags & BBF_GC_SAFE_POINT)))
```

Comment:
It is not clear to me why `BBF_GC_SAFE_POINT` would even be set on a block that ends with a tail call.
Also, since you are here, you should change these to check the value, e.g. `((topBB->bbFlags & BBF_GC_SAFE_POINT) != 0)` etc.